src/krb5_plugin/passkey/passkey_utils.c

```diff
@@ -624,9 +625,10 @@ sss_passkey_concat_credentials(char **creds,
             goto done;
         }
 
+        len = strlen(tmp_creds);
         result_creds = tmp_creds;
 
-        strncat(result_creds, creds[i], creds_len);
+        strncat(result_creds, creds[i], len);
```

Comment:
Either I miss something, or this doesn't fix a real issue.
`result_creds = strdup(creds[0]);`
and then on the very first iteration in the loop:
`strcat(result_creds, ",");`  --  this is already buffer overrun. At this point '\0' at the end of `result_creds` is overwritten by ',' and then new '\0' is added *outside* of allocated buffer.

As for `strncat()` within loop, it should specify free space left in `result_creds`, not its entire size (i.e. something like strlen(creds[i]) +- 1)
Moreover, since exact space was allocated, there is just no need to use `strncat()` - `memcpy()` will do better.

And, finally, not related to covscan report and maybe not important here, but `realloc()` is quite heavy function. Maybe not worth bothering here (I don't know how big creds array might get), but in general it would be much more efficient to calculate buffer size in a first loop, do single alloc, fill it.

Reply:
Fixed, thanks. I checked and upstream `scan.coverity.com` did not find issues with these changes also.